pkg/common/client.go

```diff
@@ -179,15 +179,18 @@ func (c *Client) FindUserByEmail(email, realm string) (*v1alpha1.KeycloakAPIUser
 }
 
 func (c *Client) FindUserByUsername(name, realm string) (*v1alpha1.KeycloakAPIUser, error) {
-	result, err := c.get(fmt.Sprintf("realms/%s/users?first=0&max=1&search=%s", realm, name), "user", func(body []byte) (T, error) {
+	result, err := c.get(fmt.Sprintf("realms/%s/users?username=%s", realm, name), "user", func(body []byte) (T, error) {
```

Comment:
This API endpoint has a max field with a default of 100.
Docs: https://www.keycloak.org/docs-api/9.0/rest-api/index.html#_getusers

We should pass it to get all results. Maybe `0` value will work.

Reply:
In Keycloak there's a unique constraint on username (see [this piece of the code](https://github.com/keycloak/keycloak/blob/b478472b3578b8980d7b5f1642e91e75d1e78d16/model/jpa/src/main/resources/META-INF/jpa-changelog-2.5.0.xml#L82)). So gathering all users doesn't make sense to me as you will either get 1 entity back or nothing.

So I would prefer to put `first=0&max=1` back or use `search` instead of `username`.

Reply:
@slaskawi Even if the username has a unique constraint, it doesn't stop it from being a substring of another username/containing another username in it. 

The problem with using `max=1` is that the API `username` or `search` filter seems to return all users that contains the specified string. The first result of this request isn't necessarily the correct user even if the username is unique.

I would agree with @stianst suggestion of adding the uid to the User CR. This way we're ensuring that the correct user is always returned especially if the username value can be changed.

Reply:
@slaskawi From the observed behavior it seemed like the API endpoint we use here^ is using this query:
https://github.com/keycloak/keycloak/blob/93a1374558643ee31e717c1dfacecb8ab8d95f68/model/jpa/src/main/java/org/keycloak/models/jpa/entities/UserEntity.java#L47-L48
and the query uses `like` instead of exact matching.

Reply:
@matskiv Looks like `max=0` doesn't disable this but rather just sets the maximum result to 0, being an empty list. I've set it to `max=-1` instead.